turbo.json

```diff
@@ -4,13 +4,18 @@
   "pipeline": {
     "lint": {},
     "lint:fix": { "cache": false },
-    "test": {},
+    "test": {
+      "dependsOn": ["^compile"]
+    },
+    "build": {
+      "dependsOn": ["^compile"]
+    },
```

Comment:
This is where the magic happens. Here we say that _"Before executing the `build` script of a package, make sure that the `compile` task OF ITS DEPENDENCIES (that's what the `^` means) is executed first"._

So, for the desktop wallet, for example, the `shared` package is listed in its dependencies array. Thus, when running `turbo run build` (or `bun run ci:build`), the `compile` script of the shared package will run first. ✨

Reply:
All clear 🫡